```
cmd/disk-cache_test.go:28:28: too many arguments in call to m.ToObjectInfo (typecheck)
	objInfo := m.ToObjectInfo("testbucket", "testobject")
	                          ^
make: *** [Makefile:38: lint] Error 1
Error: Process completed with exit code 2.
```

```
# github.com/minio/minio/cmd [github.com/minio/minio/cmd.test]
cmd/disk-cache_test.go:28:27: too many arguments in call to m.ToObjectInfo
	have (string, string)
	want ()
FAIL	github.com/minio/minio/cmd [build failed]
FAIL
make: *** [Makefile:55: test-iam] Error 2
```

@poornas

--- a/cmd/disk-cache-backend.go
+++ b/cmd/disk-cache-backend.go
@@ -411,6 +411,9 @@ func (c *diskCache) purge(ctx context.Context) {
 		}
 
 		for fname, fi := range cachedRngFiles {
+			if fi == nil {
+				continue
+			}
 			if cc != nil {
 				if cc.isStale(objInfo.ModTime) {
 					removeAll(fname)
@@ -428,9 +431,11 @@ func (c *diskCache) purge(ctx context.Context) {
 		}
 		// clean up stale cache.json files for objects that never got cached but access count was maintained in cache.json
 		fi, err := os.Stat(pathJoin(cacheDir, cacheMetaJSONFile))
-		if err != nil || (fi.ModTime().Before(expiry) && len(cachedRngFiles) == 0) {
+		if err != nil || (fi != nil && fi.ModTime().Before(expiry) && len(cachedRngFiles) == 0) {
 			removeAll(cacheDir)
-			scorer.adjustSaveBytes(-fi.Size())
+			if fi != nil {
+				scorer.adjustSaveBytes(-fi.Size())
+			}
 			// Proceed to next file.
 			return nil
 		}

--- a/cmd/disk-cache_test.go
+++ b/cmd/disk-cache_test.go
@@ -25,7 +25,7 @@ import (
 // Tests ToObjectInfo function.
 func TestCacheMetadataObjInfo(t *testing.T) {
 	m := cacheMeta{Meta: nil}
-	objInfo := m.ToObjectInfo("testbucket", "testobject")
+	objInfo := m.ToObjectInfo()
 	if objInfo.Size != 0 {
 		t.Fatal("Unexpected object info value for Size", objInfo.Size)
 	}

